lnpbp-0031.md

```diff
@@ -40,8 +42,140 @@ License: CC0-1.0
+```
+
+`Bitcoin.sty`, semantic lib id `panel_chamber_ohio_GWzzfwBqzA5BUVER6hqq3rBdDs6UUJ15w1T8ys6WrNr5`:
+```haskell
+typelib Bitcoin -- version v0.10.1-2023.04.20.B
+
+data LockTime         :: U32
+data Outpoint         :: txid Txid, vout Vout
+data Sats             :: U64
+data ScriptBytes      :: [U8 ^ ..0xffffffffffffffff]
+data ScriptPubkey     :: ScriptBytes
+data SeqNo            :: U32
+data SigScript        :: ScriptBytes
+data Tx               :: version TxVer
+                       , inputs [TxIn ^ ..0xffffffffffffffff]
+                       , outputs [TxOut ^ ..0xffffffffffffffff]
+                       , lockTime LockTime
+data TxIn             :: prevOutput Outpoint
+                       , sigScript SigScript
+                       , sequence SeqNo
+                       , witness Witness
+data TxOut            :: value Sats, scriptPubkey ScriptPubkey
+data TxVer            :: U32
+data Txid             :: [U8 ^ 32]
+data Vout             :: U32
+data Witness          :: [[U8 ^ ..0xffffffffffffffff] ^ ..0xffffffffffffffff]
+```
+
+`CommitVerify.sty`, semantic lib id `texas_year_ethnic_CPr8tcdPqWZ3KP8dXNPYavTEkbn8PG7CoJHtfwDFKRHJ`
+```haskell
+typelib CommitVerify -- version v0.10.0-2023.03.05.A
+
+data Message          :: [U8 ^ 32]
+data ProtocolId       :: [U8 ^ 32]
+data MerkleNode       :: [U8 ^ 32]
+
+data TreeNode         :: concealedNode (depth StdLib.U4, hash MerkleNode)
+                       | commitmentLeaf (protocolId ProtocolId, message Message)
+
+data MerkleTree       :: depth StdLib.U4
+                       , entropy U64
+                       , messages {[Byte ^ 32] -> Message}
+                       , map {U16 -> ProtocolId, Message}
+
+data MerkleBlock      :: depth StdLib.U4
+                       , crossSection [TreeNode]
+                       , entropy U64?
+
+data MerkleProof      :: pos U16, path [MerkleNode]
+```
+
+`RGBContracts.sty`, semantic lib id ``:
+```haskell
+typelib RGBTypes -- version v0.10.1-2023.04.20.C
+
+import panel_chamber_ohio_GWzzfwBqzA5BUVER6hqq3rBdDs6UUJ15w1T8ys6WrNr5 as Bitcoin
+import olivia_angel_micro_Fd3Lx2smvwdKhRCkhMs2du3CAYyKqJneNSf1g72WjBoq as _
+
+-- number of decimal fractions (decimal numbers after floating point)
+data Precision :: indivisible:0 
+                | deci:1 
+                | centi:2 
+                | milli:3
+                | deciMilli:4
+                | centiMilli:5 
+                | micro:6 
+                | deciMicro:7 
+                | centiMicro:8 
+                | nano:9 
+                | deciNano:10 
+                | centiNano:11 
+                | pico:12 
+                | deciPico:13 
+                | centiPico:14 
+                | femto:15 
+                | deciFemto:16 
+                | centiFemto:17 
+                | atto:18
+
+data Amount :: Zk64 -- fungible asset amount
```

Comment:
Not sure, is this a standard type or did you forget to define this type?

Reply:
It is a standard type for Contractum language